### ISSUE TITLE
Canvas fingerprinting

### ISSUE THREAD

**Issue by ekaris** [schomery/privacy-settings#57]
Add an option to prevent canvas fingerprinting:
https://addons.mozilla.org/firefox/addon/canvas-fingerprint-blocker/

**Comment by lordraiden** [comment]
This is the best alternative right now
https://addons.mozilla.org/es/firefox/addon/no-canvas-fingerprinting/

A must read
https://multiloginapp.com/how-canvas-fingerprint-blockers-make-you-easily-trackable/

**Comment by Atavic** [comment]
Experimental - it has no Privacy Policy:
https://addons.mozilla.org/en-US/firefox/addon/no-canvas-fingerprinting/reviews/815207/

**Comment by Gitoffthelawn** [comment]
I like the idea behind Canvas Defender, but when performing some code profiling, it looked like it was the cause of slowing things down a bit.  Not a huge amount, mind you... but enough to be a concern.

**Comment by Atavic** [comment]
One concern is that this addon is experimental as it doesn't reach minimum mozilla standars by not providing any privacy policy whatsoever.

**Comment by Gitoffthelawn** [comment]
@Atavic I agree, that's concerning.  It's a bit odd that there is no policy, given that it's written by a company, and they have a privacy policy for their website: https://multiloginapp.com/privacy-policy/

**Comment by LimboSlam** [comment]
Haha! There has been a much longer, living add-on which does the same thing and maybe more. It's called CanvasBlocker: https://addons.mozilla.org/en-US/firefox/addon/canvasblocker/. Now I'm pretty sure this one has a Privacy Policy, at least last time I checked.

**Comment by Gitoffthelawn** [comment]
@LimboSlam Very true, but read the second link in the second post in this thread for an article about downsides to using the technique employed by that extension.

**Comment by Atavic** [comment]
[CanvasBlocker vs. Canvas Defender](https://github.com/privacytoolsIO/privacytools.io/issues/99)

**Comment by Atavic** [comment]
I see the 2nd link on [this post](https://github.com/schomery/privacy-settings/issues/57#issuecomment-239664096) a little biased IMHO. You can counter that with [this text](https://github.com/privacytoolsIO/privacytools.io/issues/99#issuecomment-316519210), partially taken from the Tor Project.

Just pointing at more ideas about this _Randomization vs Obfuscation vs Uniformity_ strategies, as I don't look at :-1: anyone.
